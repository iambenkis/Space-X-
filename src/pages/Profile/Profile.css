li {
  list-style: none;
  font-size: 1.3rem;
}

.profile {
  font-size: 1.3rem;
  padding: 2rem;
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
  gap: 4rem;
}

.profile h3 {
  font-size: 2rem;
  font-weight: 500;
  margin-bottom: 0.5rem;
}

.rocket-profile,
.mission-profile {
  width: 50%;
}

.profile ul {
  border: 1px solid #777;
  border-bottom: none;
  width: 100%;
}

.profile li {
  font-size: 1.7rem;
  border-bottom: 1px solid #777;
  padding: 1rem;
}
