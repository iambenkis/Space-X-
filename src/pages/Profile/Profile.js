import React from 'react';
import { useSelector } from 'react-redux';
import './Profile.css';

const Profile = () => {
  const rocketStore = useSelector((state) => state.rockets);
  const missionStore = useSelector((state) => state.missions);
  const reservedRocket = rocketStore.filter((r) => r.reserved);
  const joinedMission = missionStore.payload.filter((r) => r.joined);

  return (
    <>
      <div className="profile">
        <div className="rocket-profile">
          <h3>My Rockets</h3>
          <ul>
            {
              reservedRocket.map((r) => <li key={r.id}>{r.name}</li>)
            }
          </ul>
        </div>

        <div className="mission-profile">
          <h3>My Missions</h3>
          <ul>
            {
              joinedMission.map((r) => <li key={r.mission_id}>{r.mission_name}</li>)
            }
          </ul>
        </div>
      </div>
    </>
  );
};

export default Profile;
