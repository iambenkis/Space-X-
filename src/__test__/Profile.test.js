import React from "react";
import '@testing-library/jest-dom';
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import renderer from 'react-test-renderer';
import store from "../redux/Rockets/configurationStore";
import Profile from '../pages/Profile/Profile'

describe("Test Profile component", () => {
    test("Does Profile renderer ? ", () => {
        const profileReserved = renderer.create(
            <BrowserRouter>
                <Provider store={store}>
                    <Profile />
                </Provider>
            </BrowserRouter>
        ).toJSON();
        expect(profileReserved).toMatchSnapshot();
    })
})