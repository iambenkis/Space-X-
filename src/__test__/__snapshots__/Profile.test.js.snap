// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Profile component Does Profile renderer ?  1`] = `
<div
  className="profile"
>
  <div
    className="rocket-profile"
  >
    <h3>
      My Rockets
    </h3>
    <ul />
  </div>
  <div
    className="mission-profile"
  >
    <h3>
      My Missions
    </h3>
    <ul />
  </div>
</div>
`;
